### ISSUE TITLE
Permitting comment separators in "final-definition" 

### ISSUE THREAD

**Issue by lmn8** [remarkjs/remark-lint#244]
### Subject of the feature

Allow devs to add header-like comments separating link definitions

### Problem

I am using HTML comments at EOF to separate my link definitions by version:
```markdown
<!-- Base Links -->
[gh-remark]: https://github.com/remarkjs/remark-lint
[remark-js]: https://remark.js.org/
<!-- v12.0.0 -->
[npm]: https://www.npmjs.com/package/remark/v/12.0.0
```
The first two links throw errors because I've chosen to enable `final-definition` rule..

### Expected behaviour

The links shouldn't throw an error, or there should be an option to permit the comments.

### Alternatives

Disable "final-definition" in my preset with an override (I would have to document and track this also)

**Comment by wooorm** [comment]
I think such behavior would make sense! Would you be willing to create a PR?
Here’s how we allow comments in the [`no-html` rule](https://github.com/remarkjs/remark-lint/blob/fa234a3e75d58b6d2e4e5f8a6685a49b7135cb3a/packages/remark-lint-no-html/index.js#L41).
